### PR TITLE
incorporate TLS key passphrase support from beats.

### PR DESCRIPTION
**Pull request by jalvz**
Co-authored-by: sterchelen <sterchele.nicolas@gmail.com>

### CODE REVIEW

**Comment by jalvz** [comment]
retest this please

File: apm-server.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ apm-server:
   #ssl.enabled: false
   #ssl.certificate : "path/to/cert"
   #ssl.key : "path/to/private_key"
+  #ssl.key_passphrase: ""
```
[the remark this answers is not in the document]
Comment (jalvz) [reply apm-server.yml]: I agree, but that should be in the docs? 
I mean, if there is a password option, shouldn't be commented out like the rest? We don't have any other "best practices" in the yml files

[the remark this answers is not in the document]
Comment (simitt) [reply apm-server.yml]: You're right that it should be in the docs. I would align with how we handle experimental features, where we also add a note in the yml files that it is experimental.

WDYT about adding a note like _Instead of entering the passphrase here in plain text we recommend you use the [keystore](https://www.elastic.co/guide/en/apm/server/current/keystore.html) instead_.

**Comment by jalvz** [comment]
Merging this now, thanks a lot @sterchelen  for your work on this, you are due credited as co-author.

**Pull request merged**